llama-index-integrations/readers/llama-index-readers-microsoft-sharepoint/llama_index/readers/microsoft_sharepoint/base.py

```diff
@@ -131,6 +131,40 @@ def _get_access_token(self) -> str:
             logger.error("Error retrieving access token: %s", json_response["error"])
             raise ValueError(f"Error retrieving access token: {error_message}")
 
+    def _send_request_with_retry(self, request: requests.Request) -> requests.Response:
+        """
+        Makes a request to the SharePoint API with the provided request object.
+        If the request fails with a 401 status code, the access token is refreshed and the request is retried once.
+        """
+        curr_headers = (request.headers or {}).copy()
+        curr_headers.update(self._authorization_headers)
+        request.headers = curr_headers
+        prepared_request = request.prepare()
+        with requests.Session() as session:
+            response = session.send(prepared_request)
+
+            if response.status_code == 401:
+                # refresh token
+                logger.debug("Received 401. Refreshing access token.")
```

Comment:
Why only 401 errors?

Reply:
Thats the error code we see when the `access_token` has timed out, which is the specific case I'm trying to handle with this

Reply:
Can we add a comment with those details specifically? Otherwise, looks good!